consensus/state.go

```diff
@@ -806,8 +806,14 @@ func (cs *ConsensusState) needProofBlock(height int64) bool {
 }
 
 func (cs *ConsensusState) proposalHeartbeat(height int64, round int) {
-	counter := 0
+	logger := cs.Logger.With("height", height, "round", round)
 	addr := cs.privValidator.GetAddress()
+
+	if !cs.Validators.HasAddress(addr) {
+		logger.Debug("Not sending proposalHearbeat. This node is not a validator", "addr", addr, "vals", cs.Validators)
```

Comment:
not necessary, I think

```suggestion
```

Reply:
As "proposalHeartbeat" is a debugging mechanism, the idea here is to also aid you in understanding why a broadcast has not been sent.